test/exhal/transcoder_test.exs

```diff
@@ -145,3 +145,91 @@ defmodule ExHal.TranscoderTest do
 
   end
 end
+
+defmodule ExHal.TranscoderEmptyLinksTest do
+  use ExUnit.Case
+
+  setup do
+    hal = """
+    {
+      "_links": {
+        "tag": []
+      }
+    }
+    """
+
+    {:ok, doc: ExHal.Document.parse!(ExHal.client, hal)}
+  end
+
+  test "trying to extract empty link array", %{doc: doc} do
+   defmodule MyOtherMultiLinkTranscoder do
+     use ExHal.Transcoder
+
+     deflinks "tag", param: :tag
+   end
+
+   assert %{} == MyOtherMultiLinkTranscoder.decode!(doc)
+
+   encoded = MyOtherMultiLinkTranscoder.encode!(%{tag: []})
+   refute ExHal.Document.has_link?(encoded, "tag")
+  end
+end
+
+
+defmodule ExHal.TranscoderNullLinksTest do
+  use ExUnit.Case
+
+  setup do
+    hal = """
+    {
+      "_links": {
+        "tag": null
+      }
+    }
+    """
+
+    {:ok, doc: ExHal.Document.parse!(ExHal.client, hal)}
+  end
+
+  test "trying to extract empty link array", %{doc: doc} do
+   defmodule MyOtherMultiLinkTranscoder do
+     use ExHal.Transcoder
+
+     deflinks "tag", param: :tag
+   end
+
+   assert %{} == MyOtherMultiLinkTranscoder.decode!(doc)
+
+   encoded = MyOtherMultiLinkTranscoder.encode!(%{tag: []})
+   refute ExHal.Document.has_link?(encoded, "tag")
+  end
+end
+
+defmodule ExHal.TranscoderAbsentLinksTest do
```

Comment:
Is this a duplicated of [line 149](https://github.com/pezra/exhal/pull/36/files#diff-9defe435edba51bde3e9f11bee304c3dR149)?